Simplenote/src/main/java/com/automattic/simplenote/AddTagActivity.java

```diff
@@ -15,78 +15,60 @@
 import androidx.annotation.Nullable;
 import androidx.appcompat.app.AlertDialog;
 import androidx.appcompat.app.AppCompatActivity;
+import androidx.lifecycle.Observer;
+import androidx.lifecycle.ViewModelProvider;
 
+import com.automattic.simplenote.databinding.ActivityTagAddBinding;
 import com.automattic.simplenote.models.Tag;
 import com.automattic.simplenote.utils.DisplayUtils;
 import com.automattic.simplenote.utils.HtmlCompat;
-import com.automattic.simplenote.utils.TagUtils;
 import com.automattic.simplenote.utils.ThemeUtils;
+import com.automattic.simplenote.viewmodels.AddTagViewModel;
+import com.automattic.simplenote.viewmodels.ViewModelFactory;
 import com.automattic.simplenote.widgets.MorphCircleToRectangle;
 import com.automattic.simplenote.widgets.MorphSetup;
 import com.google.android.material.textfield.TextInputEditText;
 import com.google.android.material.textfield.TextInputLayout;
 import com.simperium.client.Bucket;
-import com.simperium.client.BucketObjectNameInvalid;
 
 import java.util.Objects;
 
 public class AddTagActivity extends AppCompatActivity implements TextWatcher {
-    private Bucket<Tag> mBucketTag;
-    private Button mButtonNegative;
-    private Button mButtonPositive;
-    private TextInputEditText mTagInput;
-    private TextInputLayout mTagLayout;
+    private AddTagViewModel viewModel;
+    private ActivityTagAddBinding binding;
```

Comment:
this will cause memory leaks. Generally you need to set the binding to null in `onDestroy`. However, in this case you don't need to have this as a field at all. You can just initialize it and pass it as a parameter to all the methods because it's only used in the `onCreate` method, right?

Reply:
Fixed in ff0380dcc28559b1f4652e58177aa23988a8e8e0.